runner.go

```diff
@@ -579,7 +582,29 @@ func (r *Runner) execute(command string) error {
 	cmd.Stdout = r.outStream
 	cmd.Stderr = r.errStream
 	cmd.Env = cmdEnv
-	return cmd.Run()
+
+	done := make(chan error, 1)
+	go func() {
+		done <- cmd.Run()
+	}()
+
+	select {
+	case <-time.After(timeout):
+		if cmd.Process != nil {
+			if err := cmd.Process.Kill(); err != nil {
```

Comment:
It might be nice to try sending a SIGTERM first, and then a SIGKILL if the command still doesn't exit. It would require another signal/wait loop though, so up to you.